server/provider/urls.go

```diff
@@ -1,30 +1,50 @@
 package provider
 
 import (
+	"encoding/hex"
 	"fmt"
+	"net/url"
+	"strconv"
 
-	"github.com/ElrondNetwork/elrond-proxy-go/common"
+	"github.com/ElrondNetwork/rosetta/server/resources"
 )
 
 var (
-	urlPathGetNodeStatus           = "/node/status"
-	urlPathGetGenesisBalances      = "/network/genesis-balances"
-	urlPathGetAccount              = "/address/%s"
-	urlPathGetAccountNativeBalance = "/address/%s/balance"
-	urlPathGetAccountESDTBalance   = "/address/%s/esdt/%s"
+	urlPathGetNodeStatus                        = "/node/status"
+	urlPathGetGenesisBalances                   = "/network/genesis-balances"
+	urlPathGetAccount                           = "/address/%s"
+	urlPathGetAccountNativeBalance              = "/address/%s/balance"
+	urlPathGetAccountESDTBalance                = "/address/%s/esdt/%s"
+	urlParameterAccountQueryOptionsOnFinalBlock = "onFinalBlock"
+	urlParameterAccountQueryOptionsBlockNonce   = "blockNonce"
+	urlParameterAccountQueryOptionsBlockHash    = "blockHash"
 )
 
 func buildUrlGetAccount(address string) string {
-	options := common.AccountQueryOptions{OnFinalBlock: true}
-	return common.BuildUrlWithAccountQueryOptions(fmt.Sprintf(urlPathGetAccount, address), options)
+	options := resources.AccountQueryOptions{OnFinalBlock: true}
+	return buildUrlWithAccountQueryOptions(fmt.Sprintf(urlPathGetAccount, address), options)
 }
 
-func buildUrlGetAccountNativeBalance(address string) string {
-	options := common.AccountQueryOptions{OnFinalBlock: true}
-	return common.BuildUrlWithAccountQueryOptions(fmt.Sprintf(urlPathGetAccountNativeBalance, address), options)
+func buildUrlGetAccountNativeBalance(address string, options resources.AccountQueryOptions) string {
+	return buildUrlWithAccountQueryOptions(fmt.Sprintf(urlPathGetAccountNativeBalance, address), options)
 }
 
-func buildUrlGetAccountESDTBalance(address string, tokenIdentifier string) string {
-	options := common.AccountQueryOptions{OnFinalBlock: true}
-	return common.BuildUrlWithAccountQueryOptions(fmt.Sprintf(urlPathGetAccountESDTBalance, address, tokenIdentifier), options)
+func buildUrlGetAccountESDTBalance(address string, tokenIdentifier string, options resources.AccountQueryOptions) string {
+	return buildUrlWithAccountQueryOptions(fmt.Sprintf(urlPathGetAccountESDTBalance, address, tokenIdentifier), options)
+}
+
+func buildUrlWithAccountQueryOptions(path string, options resources.AccountQueryOptions) string {
+	u := url.URL{Path: path}
+	query := u.Query()
+
+	if options.OnFinalBlock {
```

Comment:
in order to avoid these if...else...if's, why not using a small function with early exists? The code looks better that way

Reply:
Fixed.